### DISCUSSION TITLE
Using fragment in Query

### DISCUSSION THREAD

**Discussion by Krypt0r7** [vazco/uniforms#975] [Q&A] [answered]
Hi! 
Using fragments in query doesn't seem to be working with uniforms. The Query works as for getting the data. I see no difference in the data in comparison with specify each prop in both objects.

I used <NestField name="officeAddress"> component for each object. And <AutoField name="type" /> for each prop in that object. But the data doesn't appear in the field... Any suggestions for a solution here???

Simple example of the query I used...

```
query {
  crm {
    company {
      companies {
        id
        name
        marketId
        registrationNumber
        officeAddress {
          ...Address
        }
        invoiceAddress {
          ...Address
        }
      }
    }
  }
}

fragment Address on Crm_Company_CompaniesQuery_Address {
  type
  name
  emailAddress
  phoneNumber
  address1
  address2
  postalCode
  city
  countryCode
}
```

**Comment by radekmie** [comment, marked as answer]
Hi @Krypt0r7, sorry for the delay! The fragments themselves are not really the problem here. The point is that uniforms (uniforms-bridge-graphql) do not accept queries at all. Queries are far more complex than type definitions (e.g., queries can include parameters and aliases), and thus we've decided to focus only on types.

In any case, we're open for PRs.

**Reply by Krypt0r7** [reply]
Hi @radekmie! Thanks for your reply. The query I'm using is for presenting data I already have in the input fields. OnSubmit stills fires a mutation (Crm_Company_CreateCompany). This is only for the model part. 

Using fragments, the input fields doesn't get the data. But a regular query where I specify all the props in each object does show in the inputs.

**Reply by radekmie** [reply]
@Krypt0r7 You say you use the query for getting the model, right? If so, then this issue is not related to uniforms, as uniforms have no GraphQL data-fetching functionality.

Or maybe I misunderstood something? If possible, please do include some example code.
